src/app/scenario/scenario.component.html

```diff
@@ -58,414 +58,737 @@ <h3>Scenarios</h3>
+                  id="keepalive_amount"
+                  formControlName="keepalive_amount"
+                  class="clr-input"
                 />
-                <clr-control-helper
-                  >Period in hours that a user may pause their
-                  scenario.</clr-control-helper
-                >
-                <clr-control-error *clrIfError="'required'"
-                  >Pause duration is required.</clr-control-error
-                >
-                <clr-control-error *clrIfError="'min'"
-                  >Pause duration must be 1 hour minimum.</clr-control-error
-                >
-                <clr-control-error *clrIfError="'max'"
-                  >Pause duration must be 48 hours maximum.
-                </clr-control-error>
-              </clr-input-container>
-            </form>
-            <clr-stack-view>
-              <clr-stack-header>
-                Virtual Machines
+                <select formControlName="keepalive_unit">
+                  <option value="h">Hour(s)</option>
+                  <option value="m">Minute(s)</option>
+                </select>
+              </div>
+              <clr-icon
+                class="clr-validate-icon"
+                shape="exclamation-circle"
+              ></clr-icon>
+            </div>
+            <span
+              class="clr-subtext"
+              *ngIf="
+                !keepaliveRequired &&
+                !scenarioDetails.errors?.invalidKeepalivePeriod
+              "
+              >Period before VMs are reaped upon user inactivity.</span
+            >
+            <span class="clr-subtext" *ngIf="keepaliveRequired"
+              >Keepalive duration is required.</span
+            >
+            <span
+              class="clr-subtext"
+              *ngIf="scenarioDetails.errors?.invalidKeepalivePeriod"
+              >Invalid keepalive period. Valid periods: 1m to 48h
+              inclusive.</span
+            >
+          </div>
+        </div>
+        <clr-input-container>
+          <label>Pause Duration</label>
+          <input
+            clrInput
+            type="number"
+            name="pause_duration"
+            placeholder="Duration in hours"
+            formControlName="pause_duration"
+            required
+          />
+          <clr-control-helper
+            >Period in hours that a user may pause their
+            scenario.</clr-control-helper
+          >
+          <clr-control-error *clrIfError="'required'"
+            >Pause duration is required.</clr-control-error
+          >
+          <clr-control-error *clrIfError="'min'"
+            >Pause duration must be 1 hour minimum.</clr-control-error
+          >
+          <clr-control-error *clrIfError="'max'"
+            >Pause duration must be 48 hours maximum.</clr-control-error
+          >
+        </clr-input-container>
+      </form>
+    </div>
+  </clr-wizard-page>
+
+  <clr-wizard-page
+    [clrWizardPageNextDisabled]="!isVMAdded"
```

Comment:
I would propose changing this to a function that checks if a VM is given. That way you do not need the variable.

So [clrWizardPageNextDisabled]="!scenarioHasVM()"

And scenarioHasVM() returns true if the Scenario has any VM, false if no VM is present

Reply:
Okey, I just saw it, I will implement it.

Reply:
Done